source/opt/const_folding_rules.cpp

```diff
@@ -350,14 +350,63 @@ ConstantFoldingRule FoldFPBinaryOp(BinaryScalarFoldingRule scalar_rule) {
   return [scalar_rule](IRContext* context, Instruction* inst,
                        const std::vector<const analysis::Constant*>& constants)
              -> const analysis::Constant* {
+    analysis::ConstantManager* const_mgr = context->get_constant_mgr();
+    analysis::TypeManager* type_mgr = context->get_type_mgr();
+    const analysis::Type* result_type = type_mgr->GetType(inst->type_id());
+    const analysis::Vector* vector_type = result_type->AsVector();
+
     if (!inst->IsFloatingPointFoldingAllowed()) {
       return nullptr;
     }
-    if (inst->opcode() == SpvOpExtInst) {
-      return FoldFPBinaryOp(scalar_rule, inst->type_id(),
-                            {constants[1], constants[2]}, context);
+
+    // Workraround a crash caused by DXC using null constants
+    if (constants[0] == nullptr || constants[1] == nullptr ||
+        constants[0]->AsNullConstant() || constants[1]->AsNullConstant()) {
+      return nullptr;
+    }
+
+    if (vector_type != nullptr) {
+      std::vector<const analysis::Constant*> a_components;
+      std::vector<const analysis::Constant*> b_components;
+      std::vector<const analysis::Constant*> results_components;
+
+      // Workaround a crash caused by vector binOp(scalar, vector) somehow
+      // ending up in here
```

Comment:
I would be interested in which instruction get here.  This smells like an invalid instruction in the code.